### PR TITLE
Remove setRefresh

### PR DESCRIPTION
**Pull request by nik9000**
It has been replaced with `setRefreshPolicy` which has support for
waiting until refresh with `setRefreshPolicy(WAIT_FOR)`.

Related to #1063

### CODE REVIEW

**Comment by nik9000** [comment]
Added breaking label because it is a breaking change to the Java API.

File: core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java [hunk 1]
```diff
@@ -972,7 +974,7 @@ public void testRecoverFromStore() throws IOException {
         IndexService test = indicesService.indexService(resolveIndex("test"));
         final IndexShard shard = test.getShardOrNull(0);
         int translogOps = 1;
-        client().prepareIndex("test", "test", "0").setSource("{}").setRefresh(randomBoolean()).get();
+        client().prepareIndex("test", "test", "0").setSource("{}").setRefreshPolicy(randomFrom(RefreshPolicy.values())).get();
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java]: do we want to slow down tests with wait for refreshes?

Comment (nik9000) [inline core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java]: Yeah, I think this was the wrong choice. I wanted to keep the spirit of the old test and just do a random thing, but I don't think it is worth it. Just random picking between IMMEDIATE and NONE is probably more right.

**Comment by bleskes** [comment]
LGTM. Thanks @nik9000 . No need for another review once the those test a sped up again.

**Comment by nik9000** [comment]
Thanks for reviewing @bleskes !

**Pull request merged**